### ISSUE TITLE
Vim Mode

### ISSUE THREAD

**Issue by diogox** [anyproto/anytype-ts#247]
### Have you read a contributing guide?

- [X] I have read CONTRIBUTING.md
- [X] I have searched the existing requests and didn't find any that were similar
- [X] I have considered creating a pull request instead and want to proceed

### Clear and concise description of the problem

People used to using VIM have a very hard time using anything without VIM modal editing.

### Suggested solution

Adding a `Vim Mode` option in the settings.

Other apps, like Obsidian, use the same approach.

### Additional context

It would be fantastic, if there was a way to not only add VIM keybindings, but also connect to a local neovim instance, so that we can leverage all of our VIM extensions and configs, to have a consistent experience.

I know neovim offers this kind of integration, and I believe there's a VSCode plugin that taps into it. But I've never looked into how it works.

I think regular VIM keybindings would be an amazing first step.

**Comment by nphalem** [comment]
Please put this on the roadmap, we need vim motions at least. Connect to the local Neovim instance would be even better though.

**Comment by luisliz** [comment]
Currently trying to do this.I'm doing with basic bindings for now but the neovim instance idea might be better.

**Comment by schmoelder** [comment]
@luisliz die you manage to get somewhere? I would love to give Anytype a try but vim support is a must-have for me.

**Comment by xttlegendapi** [comment]
+1

**Comment by 8-cm** [comment]
any updates on this ?

**Comment by han-steve** [comment]
This will be great. Seems like there are some typescript vim libraries. It’s harder to do on the Swift side though if we want iPad vim support.

**Comment by mawkler** [comment]
A Vim mode would be amazing!

**Comment by henrebotha** [comment]
Another vote for this.

**Comment by SmatMan** [comment]
+1 for this, would be awesome!

**Comment by remotecom** [comment]
I think basic Vim editing capabilities would be awesome.

For advanced Vim motions it's better to use Vim. A Vim plugin that allows to navigate through objects and edit them would be awesome.

Personally, I have LLMs integrated in Neovim, so combination of Anytype MCP, Claude Code and such plugin will make Anytype experience superb!

**Comment by FujiwaraChoki** [comment]
+1 for this as well. Sorry to convolute, but trying to keep this alive.

**Comment by TatuMon** [comment]
+1! Would love to see this implemented.

**Comment by Pamplemousse** [comment]
👎 !

No `vim` mode ever trumps one's setup, where one can also leverage plugins and customisation.
Adding maintenance burden for a `vim` mode that will **never** match **everyone**'s needs.

As such, IMHO, it sounds sub-optimal to spend time and energy on this, instead of being able to edit `anytype` content directly from `vim`;
I very much second https://github.com/anyproto/anytype-ts/issues/247#issuecomment-1874961039 :

> Connect to the local Neovim instance would be even better though.

**Comment by henrebotha** [comment]
I agree with @Pamplemousse that a true embedded or connected Vim instance would be better, but I completely disagree that a built-in emulation is not worth it. Emulation already transforms the paradigm from modeless to modal editing, which is honestly 80% of the benefit.

**Comment by JanUwU42** [comment]
I'd love to have a vim mode!

**Comment by d0nd3x** [comment]
I hope the dev team reconsiders this. It's the only thing holding me back from really using anytype.
